### PR TITLE
DMP-847 Share data between steps

### PR DESCRIPTION
**Pull request by trev-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-847

### Change description ###
Share data between steps & update other classes to reflect this

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/automation/utils/Postgres.java [hunk 1]
```diff
@@ -39,110 +46,259 @@ public Postgres() {
...
+				conn.rollback();
+				if (updateCount == 0) {
+					throw new Exception("No rows updated");
+				} else {
+					throw new Exception("Excess rows updated " + updateCount);
+				}
+			}
+		}
+		return updateCount;
+	}
```
Comment (AdamHall-HMCTS) [inline src/main/java/uk/gov/hmcts/darts/automation/utils/Postgres.java:245]: Would it make sense, here and elsewhere it happens in the code, once the SQL string for "updateRow" has been constructed to call the other "updateRow" method that expects a fully constructed SQL string? 
[To help avoid potential code duplication.]

File: src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+	
+	static String currentYear =  Year.now().toString();
+	Properties propFile; // = new Properties();
+	Properties properties; // = new Properties();
+	public String statusCode = "";
+	public String responseString = "";
+	
+	public TestData() {
+		propFile = new Properties();
+		properties = new Properties();
```
Comment (AdamHall-HMCTS) [inline src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java:40]: Did I miss it in the method or is this dead code now?

**Review by AdamHall-HMCTS — COMMENTED** [review]
Completed a basic review of this PR, a few comments have been added to the class files.

File: src/main/java/uk/gov/hmcts/darts/automation/utils/Database.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+			"darts.courtroom ctr\r\n" +
+			"left join darts.courthouse cth\r\n" +
+			"on cth.cth_id = ctr.cth_id\r\n";
+	
+	final String courtCaseJoin = 
+			"darts.court_case cas\r\n" +
+			"inner join darts.courthouse ctr\r\n" +
+			"on ctr.ctr_id = cas.ctr_id";
+	
+	final String eventjoin =
```
[the remark this answers is not in the document]
Comment (trev-hmcts) [reply src/main/java/uk/gov/hmcts/darts/automation/utils/Database.java:37]: yep - done

File: src/main/java/uk/gov/hmcts/darts/automation/utils/JsonString.java [hunk 4]
```diff
@@ -19,7 +19,7 @@ public JsonString() {
 * 
 */
 	public JsonString addJsonLine(String tag, String value) {
-		if (!value.isBlank()) {
+		if (value != null && !value.isBlank()) {
 			if (value.equalsIgnoreCase("BLANK") || value.equalsIgnoreCase("EMPTY")) {
```
[the remark this answers is not in the document]
Comment (trev-hmcts) [reply src/main/java/uk/gov/hmcts/darts/automation/utils/JsonString.java:23]: part of wider sharing data between steps

File: src/main/java/uk/gov/hmcts/darts/automation/utils/Postgres.java (hunk 1 above)
Comment (trev-hmcts) [reply src/main/java/uk/gov/hmcts/darts/automation/utils/Postgres.java:245]: yep
Done

File: src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java (hunk 2 above)
Comment (trev-hmcts) [reply src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java:40]: looks good to me

File: src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+ */
+	
+	void saveProperty(String property, String value) {
+		properties.setProperty(property, value);
+		propFile.setProperty(property, value);
+		
+		try {
+			propFile.store(new FileWriter(parameterFileName), "values to substitute in scenarios");
+		} catch (IOException e) {
+			Assertions.fail("Error saving proprty: " + property);
```
[the remark this answers is not in the document]
Comment (trev-hmcts) [reply src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java:78]: :-)

File: src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+			if (returnValue == null) {
+				returnValue = readProperty(property);
+				if (returnValue != null) {
+					switch(property) {
+						case "case_number": 
+							String courtHouse = getProperty("courthouse");
+							if (!courtHouse.isBlank()) {
+								String caseType = getProperty("case_type");
+								if (!caseType.isBlank()) {
+									returnValue = getNextCaseNumber(getProperty("courthouse"), getProperty("case_tyoe"));
```
[the remark this answers is not in the document]
Comment (trev-hmcts) [reply src/main/java/uk/gov/hmcts/darts/automation/utils/TestData.java:102]: yep

File: src/main/java/log4j2.properties [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+appender.rolling.layout.pattern=[%-5level] %d{yyyy-MM-dd HH:mm:ss.SSS} [%t] %c{1} - %msg%n
+appender.rolling.filter.threshold.type = ThresholdFilter
+appender.rolling.filter.threshold.level = info
+appender.rolling.append = true
+appender.rolling.fileName = selenium.log
+appender.rolling.filePattern = selenium.log%i
+appender.rolling.policies.type = Policies
+#appender.rolling.policies.time.type = TimeBasedTriggeringPolicy
+#appender.rolling.policies.time.interval = 2
+#appender.rolling.policies.time.modulate = true
```
[the remark this answers is not in the document]
Comment (trev-hmcts) [reply src/main/java/log4j2.properties:18]: added comment

**Review by prekshajainHMCTS — APPROVED** [review]

**Pull request merged**